x/dualstaking/keeper/grpc_query_delegator_providers_test.go

```diff
@@ -0,0 +1,59 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/testutil/common"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDelegatorProvidersQuery(t *testing.T) {
+	ts := newTester(t)
+
+	ts.setupForDelegation(1, 1, 0, 0) // 1 delegator, 1 staked provider
+
+	_, delegator := ts.GetAccount(common.CONSUMER, 0)
+	_, provider := ts.GetAccount(common.PROVIDER, 0)
+
+	spec := ts.Spec("mock")
+
+	amountUint64 := uint64(100)
+	amount := sdk.NewCoin(epochstoragetypes.TokenDenom, sdk.NewIntFromUint64(amountUint64))
+
+	// delegate and query
+	_, err := ts.TxDualstakingDelegate(delegator, provider, spec.Index, amount)
```

Comment:
please also test:
- one provider with multiple delegator, with multiple chains
- one delegator with more than one provider
- one delegator to unstaking, and to unstaked, provider.

Reply:
fixed unit tests. see https://github.com/lavanet/lava/commit/8a4c37623f94c6d470444b0e3edde0db7168d081 for final version